sycl/test/basic_tests/buffer/buffer_interop.cpp

```diff
@@ -79,5 +79,70 @@ int main() {
       }
     }
   }
+  // Check set_final_data
+  {
+    const size_t Size = 32;
+    int Init[Size] = {5};
+    int Result[Size] = {5};
+    cl_int Error = CL_SUCCESS;
+
+    queue MyQueue;
+
+    cl_mem OpenCLBuffer = clCreateBuffer(
+        MyQueue.get_context().get(), CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+        Size * sizeof(int), Init, &Error);
+    CHECK_OCL_CODE(Error);
+    {
+      buffer<int, 1> Buffer(OpenCLBuffer, MyQueue.get_context());
+      Buffer.set_final_data(Result);
+
+      MyQueue.submit([&](handler &CGH) {
+        auto B = Buffer.get_access<access::mode::write>(CGH);
+        CGH.parallel_for<class FinalData>(
+            range<1>{Size}, [=](id<1> Index) { B[Index] = 10; });
+      });
+    }
+    Error = clReleaseMemObject(OpenCLBuffer);
+    CHECK_OCL_CODE(Error);
+    for (size_t i = 0; i < Size; ++i) {
+      if (Result[i] != 10) {
+        std::cout << " array[" << i << "] is " << Result[i] << " expected "
+                  << 10 << std::endl;
+        assert(false);
```

Comment:
Do you mean `std::terminate()` here?
Otherwise the `assert()` will be removed when compiled with `-DNDEBUG` (think Release mode for example...).

Reply:
I just write test same as other tests. I also don't like asserts so often I add Failed variable and return in main. I'm not sure that I want to add std::terminate() here because in other tests we have asserts... I think we can change tests to use something instead of asserts but not in this patch :)

Reply:
+1.
Ronan, these tests are not compiled with -DNDEBUG, so asserts are not removed. 

If you think it should be de-factored, could you file a separate issue, please? It's common practice for all the tests, not only this one.

Reply:
While I prefer `std::terminate()`, I am fine if it is the way to write test cases... Actually `assert()` might provide more information such as the file, line number, etc. when it fails.